packages/taquito-core/src/errors.ts

```diff
@@ -0,0 +1,254 @@
+// ==========================================================================================
+// parent error classes for Taquito
+// ==========================================================================================
+/**
+ *  @category Error
+ *  @description Parent error class all taquito errors to extend from
+ */
+export class TaquitoError extends Error {}
+
+/**
+ *  @category Error
+ *  @description Error indicates invalid user inputs
+ */
+export class ParameterValidationError extends TaquitoError {}
+
+/**
+ *  @category Error
+ *  @description Error returned by RPC node
+ */
+export class RpcError extends TaquitoError {}
+
+/**
+ *  @category Error
+ *  @description Error indicates TezosToolKit has not been configured appropriately
+ */
+export class TezosToolkitConfigError extends TaquitoError {}
+
+/**
+ *  @category Error
+ *  @description Error indicates a requested action is not supported by Taquito
+ */
+export class UnsupportedAction extends TaquitoError {}
+
+/**
+ *  @category Error
+ *  @description Error returned by network
+ */
+export class NetworkError extends TaquitoError {}
+
+/**
+ *  @category Error
+ *  @description Error indicates user attempts an action without necessary permissions
+ */
+export class PermissionDeniedError extends TaquitoError {}
+
+// ==========================================================================================
+// common error classes for Taquito
+// ==========================================================================================
+/**
+ *  @category Error
+ *  @description Error indicates an invalid originated or implicit address being passed or used
+ */
+export class InvalidAddressError extends ParameterValidationError {
```

Comment:
The address validation also returns a value of 
```
export enum ValidationResult {
  NO_PREFIX_MATCHED,
  INVALID_CHECKSUM,
  INVALID_LENGTH,
  VALID,
}
```
This error can accept that and expose it, so the calling code can act based on the type of error (like giving the user appropriate feedback)

Reply:
Yes, I wrote a helper function in @taquito-utils/validators.ts  called invalidErrorDetail will read the validation result and generate a string accordingly for errorDetail to display what was wrong